trieste/models/gpflow/sampler.py

```diff
@@ -795,24 +798,32 @@ def __init__(
         :param n_components: The desired number of features.
         """
 
+        super().__init__(model, n_components)
+
         if isinstance(model, SupportsGetInducingVariables):
-            inducing_points = model.get_inducing_variables()[0]  # [M, D]
+            self._inducing_points = model.get_inducing_variables()[0]  # [M, D]
         else:
-            inducing_points = model.get_internal_data().query_points  # [M, D]
+            self._inducing_points = model.get_internal_data().query_points  # [M, D]
 
-        self._cannonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
-            model.get_kernel().K(inducing_points, x)
-        )
-
-        super().__init__(model, n_components)
+        if self.is_multioutput:
+            # mypy is confused and doesn't see that kernel can be of two types
+            kernel: MultioutputKernel = model.get_kernel()  # help mypy
+            self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
+                kernel.K(self._inducing_points, x, full_output_cov=False)
+            )
+        else:
+            kernel = model.get_kernel()
+            self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
+                kernel.K(self._inducing_points, x)
+            )
 
-    def __call__(self, x: TensorType) -> TensorType:  # [N,D] -> [N, L + M]
+    def call(self, x: TensorType) -> TensorType:  # [N, D] -> [N, L + M] or [P, N, L + M]
```

Comment:
don't get this change, a bit of education please :)

Reply:
This is a keras layer and the [recommended](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#args_1) methods to override are `build` (for initialisation that depends on input shape) and `call`, instead of directly overriding `__call__`. This means the dummy call in  `ResampleableRandomFourierFeatureFunctions.__init__` can also be removed (as the keras layer will automatically build the weights if required before the call).